eigentrust-zk/src/ecdsa/mod.rs

```diff
@@ -431,17 +432,191 @@ where
+	s_inv: UnassignedInteger<C::ScalarExt, N, NUM_LIMBS, NUM_BITS, P>,
+}
+
+impl<C: CurveAffine, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P, EC>
+	EcdsaAssigner<C, N, NUM_LIMBS, NUM_BITS, P, EC>
+where
+	P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS> + RnsParams<C::ScalarExt, N, NUM_LIMBS, NUM_BITS>,
+	EC: EccParams<C>,
+	C::Base: FieldExt,
+	C::ScalarExt: FieldExt,
+{
+	/// Creates a new ecdsa assigner object
+	pub fn new(
+		public_key: UnassignedPublicKey<C, N, NUM_LIMBS, NUM_BITS, P, EC>,
+		g_as_ecpoint: UnassignedEcPoint<C, N, NUM_LIMBS, NUM_BITS, P, EC>,
+		signature: UnassignedSignature<C, N, NUM_LIMBS, NUM_BITS, P>,
+		msg_hash: UnassignedInteger<C::ScalarExt, N, NUM_LIMBS, NUM_BITS, P>,
+		s_inv: UnassignedInteger<C::ScalarExt, N, NUM_LIMBS, NUM_BITS, P>,
+	) -> Self {
+		Self { public_key, g_as_ecpoint, signature, msg_hash, s_inv }
+	}
+}
+
+impl<C: CurveAffine, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P, EC> Chipset<N>
+	for EcdsaAssigner<C, N, NUM_LIMBS, NUM_BITS, P, EC>
+where
+	P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS> + RnsParams<C::Scalar, N, NUM_LIMBS, NUM_BITS>,
+	EC: EccParams<C>,
+	C::Base: FieldExt,
+	C::ScalarExt: FieldExt,
+{
+	type Config = EcdsaAssignerConfig;
+	type Output = AssignedEcdsa<C, N, NUM_LIMBS, NUM_BITS, P, EC>;
+
+	fn synthesize(
+		self, common: &CommonConfig, config: &Self::Config, mut layouter: impl Layouter<N>,
+	) -> Result<Self::Output, Error> {
+		let aux_assigner = AuxAssigner::<C, N, NUM_LIMBS, NUM_BITS, P, EC>::new();
+		let auxes =
+			aux_assigner.synthesize(common, &config.aux, layouter.namespace(|| "aux assigner"))?;
+
+		let public_key_assigner = PublicKeyAssigner::new(self.public_key.clone());
+		let public_key = public_key_assigner.synthesize(
+			common,
+			&(),
+			layouter.namespace(|| "public_key assigner"),
+		)?;
+
+		let g_as_ecpoint_assigner = PointAssigner::new(self.g_as_ecpoint.clone());
+		let g_as_ecpoint = g_as_ecpoint_assigner.synthesize(
+			common,
+			&(),
+			layouter.namespace(|| "g_as_ec_point assigner"),
+		)?;
+
+		let signature_assigner = SignatureAssigner::new(self.signature.clone());
+		let signature = signature_assigner.synthesize(
+			common,
+			&(),
+			layouter.namespace(|| "signature assigner"),
+		)?;
+
+		let msg_hash_assigner = IntegerAssigner::new(self.msg_hash.clone());
+		let msg_hash = msg_hash_assigner.synthesize(
+			common,
+			&(),
+			layouter.namespace(|| "msg_hash assigner"),
+		)?;
+
+		let is_one = self.s_inv.integer.mul(&self.signature.s.integer);
+		layouter.assign_region(
+			|| "constraint for the s_inv",
+			|region: Region<'_, N>| {
+				let mut ctx = RegionCtx::new(region, 0);
+				let assign_res = ctx.assign_advice(
+					common.advice[0],
+					Value::known(
+						<P as RnsParams<C::Scalar, N, NUM_LIMBS, NUM_BITS>>::compose(
+							is_one.result.limbs,
+						),
+					),
+				)?;
+				let one = ctx.assign_fixed(common.fixed[0], N::ONE)?;
+				ctx.constrain_equal(assign_res, one)?;
```

Comment:
Here we are just assigning a number to `advice[0]` and a number to `fixed[0]` and checking if they are the same, but we are not using these numbers anywhere.

The correct solution is to pass `s: UnassignedInteger` and `s_inv: UnassignedInteger` and then:
1) Make AssignedInteger's from them
2) Multiply them using `IntegerMulChip`
3) Assign AssignedInteger that holds value of 1
3) Constrain the AssignedInteger from step 2. to be equal to AssignedInteger from step 3.

Reply:
Okay, done. Somehow I was thinking `mul()` from native is the circuit multiplication. Thank you.